db/pebble/db.go

```diff
@@ -36,11 +36,11 @@ func New(path string) (db.DB, error) {
 	return newPebble(path, nil)
 }
 
-func NewWithOptions(path string, cacheSizeMB uint, maxOpenFiles int, colouredLogger bool) (db.DB, error) {
+func NewWithOptions(path string, cacheSizeMB uint, maxOpenFiles int, colouredLogger bool, logEncoding utils.LogEncoding) (db.DB, error) {
```

Comment:
I think it would be better to pass logger instead of passing two settings for it's initialization. So proposed signature is:
```
func NewWithOptions(path string, cacheSizeMB uint, maxOpenFiles int, log utils.Logger) (db.DB, error) {
```
from the caller side it would be:
```
} else {
                dbLogger, err := utils.NewZapLogger(...)
                // handling of err above
		database, err = pebble.NewWithOptions(cfg.DatabasePath, cfg.DBCacheSize, cfg.DBMaxHandles, cfg.Colour)
	}
```

Reply:
Addressed.